src/plugins/RepairMan.cpp

```diff
@@ -53,3 +95,18 @@ void RepairMan::OnUnitDestroyed(const sc2::Unit* unit_, Builder* builder_) {
             return;
     }
 }
+
+void RepairMan::AssignRepairTask(const sc2::Unit& damaged_) {
+    std::function<bool(const Worker&)> moreThanHalfHealth =
+    [](const Worker& worker) {
+        const sc2::Unit* unit = gAPI->observer().GetUnit(worker.Tag());
```

Comment:
THis looks strange. Why do you try to check health of a worker?

Reply:
Had to add filter functions to GetClosest methods to prevent trying to grab the same damaged worker that I was sending away. If I send away a worker to minerals with health <=50% and Free them, the GetClosestFree will just re-grab the same worker (or possibly another worker <=50% health). So the <=50% part doesn't work without the filter.

Reply:
Oh, I see now. You thought I meant only repair when the Building gets below 50%. I meant retreating the Worker when worker.health<=50% so it will not die. I can add it so Buildings don't get repairers until they are <=50% so it doesn't send 3 workers when a building gets only a single hit of damage.

Reply:
Well, it is up to you. I would ignore the buildings until some point (e.g. 50 or 75 percent of damage, 75 could be even better), but repairing after first damage event makes sense, too.

I would totally ignore worker health as this can be easily hardcoded in unit's micro for a more specific way (e.g. ask a worker to run away if it's hp is too low).  Also, it allows to force worker to retreat in other situations and can be generalised for other units.

Reply:
Ok, I'll ignore Worker health for now and later in Unit micro, create RETREATING job behavior.